rust/lance/src/index/vector/ivf/io.rs

```diff
@@ -293,6 +294,7 @@ pub(super) async fn write_hnsw_index_partitions(
     let mut aux_part_files = Vec::with_capacity(ivf.num_partitions());
     let tmp_part_dir = Path::from_filesystem_path(TempDir::new()?)?;
     let mut tasks = Vec::with_capacity(ivf.num_partitions());
+    let sem = Arc::new(Semaphore::new(HNSW_PARTITIONS_BUILD_PARRALLEL));
```

Comment:
Maybe pass this as HnswBuildParameter?

Reply:
that's an idea! this actually depends on how large the dataset is